CRM/Core/BAO/CustomField.php

```diff
@@ -1629,6 +1628,14 @@ public static function formatCustomField(
       }
     }
 
+    if ($customFields[$customFieldId]['html_type'] == 'Autocomplete-Select'
+      && $customFields[$customFieldId]['data_type'] != 'ContactReference'
+      && isset($value)
```

Comment:
So again this is not checking the attributes prior to serializing the field. Also I see no particular reason to exclude ContactReference fields from being multivalued. They use the same select2 widget.

Reply:
Hi @colemanw 

Multi-select Contact ref is not handled in CiviCRM custom field ATM even though you set multiple in attributes. Hence i added condition to ignore for ContactReference field.

Thanks
Pradeep

Reply:
I think we'd be better to add 'serialize' to the metadata & run off that rather than have our 'if's on custom field type